packages/agents/src/agents/Chameleon/index.ts

```diff
@@ -52,15 +51,15 @@ export class ChameleonAgent extends NewAgent<GoalRunArgs> {
     const lastMessage = messages.slice(-1)[0];
 
     if (isLargeMsg(lastMessage)) {
-      await shortenMessage(lastMessage, this.lastQuery!, this.context);
+      await this.shortenMessage(lastMessage, this.lastQuery!);
```

Comment:
It might happen that the goal (which would be `lastMessage` in this scenario) is large enough to enter in this if and `this.lastQuery` attribute not be defined yet

Reply:
Yep, but I think we can tackle that in another PR